tests/test_models.py

```diff
@@ -433,504 +337,538 @@ def test_PatchDataset_crash():
+        CNNPatchPredictor(pretrained_model="secret_model-kather100k")
 
-def test_patch_predictor_alexnet_kather100K(_sample_patch1, _sample_patch2):
-    """Test for patch predictor with alexnet on Kather 100K dataset."""
-    # API 1, also test with return_labels
-    dataset = PatchDataset(
-        [pathlib.Path(_sample_patch1), pathlib.Path(_sample_patch2)], return_labels=True
-    )
-    probabilities_check = [1.0, 0.9999735355377197]
-    predictions_check = [6, 3]
-    _test_outputs_api1(
-        dataset,
-        "alexnet-kather100k",
-        return_probabilities=True,
-        return_labels=True,
-        probabilities_check=probabilities_check,
-        predictions_check=predictions_check,
-    )
+    # provide wrong model of unknown type, deprecated later with type hint
+    with pytest.raises(ValueError, match=r".*must be a string.*"):
+        CNNPatchPredictor(pretrained_model=123)
 
+    # test predict crash
+    predictor = CNNPatchPredictor(pretrained_model="resnet18-kather100k", batch_size=32)
 
-def test_patch_predictor_resnet34_kather100K(_sample_patch1, _sample_patch2):
-    """Test for patch predictor with resnet34 on Kather 100K dataset."""
-    # API 1, also test with return_labels
-    dataset = PatchDataset(
-        [pathlib.Path(_sample_patch1), pathlib.Path(_sample_patch2)], return_labels=True
-    )
-    probabilities_check = [1.0, 0.9979840517044067]
-    predictions_check = [6, 3]
-    _test_outputs_api1(
-        dataset,
-        "resnet34-kather100k",
-        return_probabilities=True,
-        return_labels=True,
-        probabilities_check=probabilities_check,
-        predictions_check=predictions_check,
-    )
+    with pytest.raises(ValueError, match=r".*not a valid mode.*"):
+        predictor.predict("aaa", mode="random")
+    with pytest.raises(ValueError, match=r".*must be a list of file paths.*"):
+        predictor.predict("aaa", mode="wsi")
+    with pytest.raises(ValueError):
+        predictor.predict([1, 2, 3], label_list=[1, 2], mode="patch")
+    # remove previously generated data
+    if os.path.exists("output"):
+        shutil.rmtree("output", ignore_errors=True)
 
 
-def test_patch_predictor_resnet50_kather100K(_sample_patch1, _sample_patch2):
-    """Test for patch predictor with resnet50 on Kather 100K dataset."""
-    # API 1, also test with return_labels
-    dataset = PatchDataset(
-        [pathlib.Path(_sample_patch1), pathlib.Path(_sample_patch2)], return_labels=True
-    )
-    probabilities_check = [1.0, 0.9999986886978149]
-    predictions_check = [6, 3]
-    _test_outputs_api1(
-        dataset,
-        "resnet50-kather100k",
-        return_probabilities=True,
-        return_labels=True,
-        probabilities_check=probabilities_check,
-        predictions_check=predictions_check,
+def test_patch_predictor_api(_sample_patch1, _sample_patch2):
+    """Helper function to get the model output using API 1."""
+    # must wrap or sthg stupid happens
```

Comment:
something stupid? too informal - please give more details

Reply:
For some reason the str going from the dict isnt considered as a path on local machine when running the test. Hence all the input paths to the test are wrapped again via pathlib.Path. It took me awhile to find out back then.